### ISSUE TITLE
Issues with Prehistorik and Prehistorik 2

### ISSUE THREAD

**Issue by vicokoby** [Baron-von-Riedesel/VSBHDA#15]
Prehistorik can't detect Sound blaster with VSBHDA.

Here the game: https://www.old-games.ru/game/106.html

Prehistorik 2 crashes with VSBHDA.

Here the game: https://www.old-games.ru/game/107.html

**Comment by Baron-von-Riedesel** [comment]
I'm probably not so much interested in the games itself, but the cracking intro for Prehistorik 2 is interesting - that should really work with VSBHDA.

**Comment by Baron-von-Riedesel** [comment]
Ok, I got the intro sound to work.

It's FM music that doesn't need interrupts to work. So the program masks all 16 IRQs. OTOH, SBEMU and VSBHDA need the real sound hardware interrupt to always work, for both digital sound and FM sound. And I forgot to protect the master PIC, bit 2 - no interrupt, no music. After protecting both master and slave PIC mask ports ( 0x21 and 0xA1 ), the sound did appear.

The fix isn't uploaded yet.

Btw, the program itself doesn't really "crash", actually sound even works to some extend, it's overloaded with a permanent humming, though - and better don't press ESC...

**Comment by vicokoby** [comment]
@Baron-von-Riedesel I tested the latest version of BANDA with Prehistorik 2 and I can confirm that the game no longer crashes. However Prehistorik 1 still cannot detect Sound Blaster.

**Comment by Baron-von-Riedesel** [comment]
> However Prehistorik 1 still cannot detect Sound Blaster.

True. However, IIRC, when the game runs in DosBox it also has problems detecting SB. Probably due to the usage of file CT-VOICE.DRV, which is a file supplied by Creative - it may use undocumented "features".
